pkg/tsdb/loki/api.go

```diff
@@ -51,7 +51,13 @@ func makeRequest(ctx context.Context, lokiDsUrl string, query lokiQuery) (*http.
 			// is ignored, so it would be nicer to not send it in such cases,
 			// but we cannot detect that situation, so we always send it.
 			// it should not break anything.
-			qs.Set("step", query.Step.String())
+			// NOTE2: we do this at millisecond precision for two reasons:
+			//  a. Loki cannot do steps with better precision anyway,
+			//     so the microsecond & nanosecond part can be ignored.
+			//  b. having it always be number+'ms' makes it more robust,
+			//     otherwise, if we want to have steps like `5m20s`,
+			//     we risk that we create a combination that Loki cannot read.
```

Comment:
thanks, i committed this, but then modified it a little to explain the problem is float-number + time-specifier, like `1.5s`, because technically, float numbers work, if you don't write anything after them, like `1.5` 👍